### ISSUE TITLE
[BUG] M600 is triggered without it being included in the gcode file

### ISSUE THREAD

**Issue by qwewer0** [MarlinFirmware/Marlin#18007]
### Short Description
Running a gcode file triggers M600 command without it being included in the gcode file.

### Bug Description

When running a non printable gcode file (e.g. [Auto0.g.zip](https://github.com/MarlinFirmware/Marlin/files/4735673/Auto0.g.zip), [Portal Still Alive M300.gcode.zip](https://github.com/MarlinFirmware/Marlin/files/4735753/Portal.Still.Alive.M300.gcode.zip)) from SD card, it triggers an M600 command without an M600 being included in the gcode files, and as a result getting an "M600: Too Cold" message on the LCD screen, and if `HOME_BEFORE_FILAMENT_CHANGE` is enabled in the firmware, then the printer will home too.
[Video of the Auto0.g file](https://imgur.com/0J2wzCP)

1. **Auto.g**: Safety backup of the M503 results.
2. **Portal Still Alive M300.gcode**: ~9 second of M300 tone.

### Performed Safety Measurements

- Tested with two different SD cards (8GB, 128MB), which are tested for errors before and after a file is copied to it.
- Every gcode and firmware file is compared to the original file (bit by bit), right after the file was copied to the SD card and after it has been removed and reinserted.
- Tested firmwares configurations:
    - UBL enabled (**277612 bytes**)
    - UBL disabled (**240044 bytes**)

### My Configurations

SKR Mini E3 v1.2, Ender 3
Latest Bugfix-2.0.x
[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/4735812/Configuration.zip)

**Comment by thisiskeithb** [comment]
Is your hotend up to temp/above `EXTRUDE_MINTEMP` before issuing a filament change?

**Comment by qwewer0** [comment]
Am I issuing a filament change? But no, the hotend is at room temperature.

With `HOME_BEFORE_FILAMENT_CHANGE` disabled this issue doesn't happens, only "M600: Too Cold" message on the LCD after the end of the file, but not always.

**Comment by thisiskeithb** [comment]
> Am I issuing a filament change? But no, the hotend is at room temperature.

[`M600`](https://marlinfw.org/docs/gcode/M600.html) is the command for filament change, so your hotend will need to be up temp. That’s why you’re getting the “too cold” message.

**Comment by qwewer0** [comment]
Even if there is no M600 in the Test.g file?

**Comment by thisiskeithb** [comment]
If you’re not issuing `M600`, then `HOME_BEFORE_FILAMENT_CHANGE` shouldn’t come into play.

> With HOME_BEFORE_FILAMENT_CHANGE disabled this issue doesn't happens, only "M600: Too Cold" message on the LCD after the end of the file, but not always.

Can you clarify “but not always”? Is this with the same exact gcode printed over and over again? It sounds intermittent like your SD card is going bad and corrupting gcode/commands if results are not consistent.

Try using a different SD card and reslicing to see if you can consistently cause the problem to occur.

**Comment by qwewer0** [comment]
It always or never happens with a specific firmware. (Minor changes in between)
But (if we can) let's ignore the "M600: Too Cold" message part, as it happened before this issue, but never caused any problem.

The main thing is that the printer 
> go into "Print Paused" and to home the printer

when using the Test.g file.

The SD card should be fine, as the file on it is (bit by bit) the same as it is in my PC.

Edit: It happens with a different SD card too.

**Comment by thisiskeithb** [comment]
The issue is intermittent, so you need to narrow down how and what causes it (without changing variables) so it’s repeatable for proper troubleshooting.

**Comment by qwewer0** [comment]
When I enable `HOME_BEFORE_FILAMENT_CHANGE` it (go into "Print Paused" and to home the printer) happens, and when I disable `HOME_BEFORE_FILAMENT_CHANGE` it doesn't happens. I recompiled, reflashed and re tested 5 times, with the same test gcode and got the same results.

Test gcode content:
```
M502						; Factory Reset
M92 X80 Y80 Z400 E137.7				; Set Axis Steps-per-unit
M201 X3000.00 Y3000.00 Z100.00 E10000.00	; Print Max Acceleration
M203 X200.00 Y200.00 Z20.00 E60.00		; Max Feedrate
M204 P500.00 R1000.00 T750.00			; Starting Acceleration
M205 J0.08 E15					; Advanced Settings
```

Let's forgot about the M600 message at the end, as for that I need to dig deeper into my firmware settings.

**Comment by thisiskeithb** [comment]
Can you post the full gcode you're trying to print instead of the test snippet?

Also, update to the latest `bugfix-2.0.x` to see if your issue persists.

**Comment by qwewer0** [comment]
The full gcode: [Auto0.zip](https://github.com/MarlinFirmware/Marlin/files/4638445/Auto0.zip)
I'm not trying to print it, as these are just my settings in a single gcode file.

I'm using the latest bugfix (updated the original description too)

Edit: Tested it on 2.0.5.3, with the same results.

**Comment by qwewer0** [comment]
<s>At first it seems to be fixed at the latest 05.18. bugfix.</s>

**Comment by qwewer0** [comment]
It seems that the issue is still present, but now I made the issue to be more general.

**Comment by boelle** [comment]
@qwewer0 still an issue?

**Comment by qwewer0** [comment]
@boelle Yes, M600 is still triggering without it being present in the gcode.
Edit: Tested on the latest bugfix

**Comment by sjasonsmith** [comment]
I want to check my understanding:

To reproduce this, all you do is "print" your Auto0.g file from the SD card? Repeat until the error occurs?

**Comment by qwewer0** [comment]
> To reproduce this, all you do is "print" your Auto0.g file from the SD card? Repeat until the error occurs?

Yes, nothing else.

**Comment by qwewer0** [comment]
Same results on Bugfix-2.0.x af8be58d.

**Comment by qwewer0** [comment]
Closing this issue, as it seems to be solved on the latest Bugfix-2.0.x 6b0d1549, but this happened already, so we will see.
